### ISSUE TITLE
Error: 'dart:ffi' can't be imported when compiling to Wasm

### ISSUE THREAD

**Issue by abrarmalekji345** [DavBfr/dart_pdf#1796]
**Describe the bug**
Can't run Flutter web WASM app after updating Flutter version to 3.27.1

**Expected behavior**
Should run Flutter web app compiled with Web Assembly 

**Screenshots**
![Screenshot 2024-12-18 at 3 55 49 PM](https://github.com/user-attachments/assets/83e8b472-0f57-4220-82f5-82e54fc7487e)

**Flutter Doctor**
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.27.1, on macOS 14.5 23F79 darwin-arm64, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.2)
[✓] Network resources

• No issues found!

**Desktop (please complete the following information):**
 - [ ] iOS
 - [ ] Android
 - [x] Browser
 - [ ] Windows
 - [ ] Linux

**Comment by navarrete0** [comment]
Hello, what do we need to continue with this? thank you.

**Comment by abrarmalekji345** [comment]
@navarrete0 For now I had to comment printing package on pubspec.yaml to run on web-assembly.

**Comment by vladimir-boyko** [comment]
The issue still persists and is a blocker for using wasm in flutter web app

**Comment by mikeesouth** [comment]
Yeah, same here - this is one of three blockers for us when moving to wasm. Commenting it for web assembly isn't really a fix for us as the pdf package is an important part of our offering. 

@DavBfr do you have any plans on supporting dart:ffi ? I haven't investigated how this package uses ffi but maybe https://pub.dev/packages/universal_ffi could be a simple solution?

**Comment by vladimir-boyko** [comment]
> Yeah, same here - this is one of three blockers for us when moving to wasm. Commenting it for web assembly isn't really a fix for us as the pdf package is an important part of our offering.
> 
> [@DavBfr](https://github.com/DavBfr) do you have any plans on supporting dart:ffi ? I haven't investigated how this package uses ffi but maybe https://pub.dev/packages/universal_ffi could be a simple solution?

Creating fork and replacing ffi with universal_ffi did help to compile library for wasm

**Comment by vladimir-boyko** [comment]
Opened a pr to resolve issue: https://github.com/DavBfr/dart_pdf/pull/1812

**Comment by karelklic** [comment]
This issue has been resolved AFAICT. Apps using the printing package can be compiled into WASM without any problem.
